File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] heights) {
        int width,height,lp=0,rp=heights.length-1,mw=0,area=0;
        while(lp<rp)
        {
            width=rp-lp;
            height=Math.min(heights[lp],heights[rp]);
            area=width*height;
            mw=Math.max(mw,area);
            if(heights[lp]<heights[rp])
                lp+=1;
            else
                rp-=1;
        }
        return mw;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, heights):
        """
        :type height: List[int]
        :rtype: int
        """
        lp=0
        rp=int(len(heights)-1)
        mw=0
        height=0
        width=0
        while lp<rp:
            width=int(rp-lp)
            height=min(heights[lp],heights[rp])
            area=int(width*height)
            mw=max(mw,area)
            if heights[lp]<heights[rp]:
                lp+=1
            else:
                rp-=1
        return mw
        